cpp_exercise/src/lib/ipcmsgqlib.cc

```diff
@@ -17,8 +17,7 @@ MsgQueueSender::MsgQueueSender(const std::string & filename): _file_name(filenam
         //opening the msg queue
         _msg_queue = mq_open(msg_queue_name.c_str(), O_WRONLY | O_CREAT , S_IRWXU | S_IRWXG, &_attrs);
         if(_msg_queue<0){
-                std::string errormsg = std::string(strerror(errno)) + "   IPCSender ERROR: Cannot open the MSGQueue.   " ;
-                throw IPCException(errormsg.c_str());
+           throw IPCException(" IPCSender ERROR: Cannot open the MSGQueue. " + std::string(strerror(errno)));
```

Comment:
As this method is the constructor. If you throw in it the destructor of MsgQueueSender won't be called. So the queue won't be closed and unlinked

Reply:
(this comment can be copied for each constructors)

Reply:
Well since it is the only function that create the msgqueue so even if it throws an exception, then no msgqueue would be created. so there is no need for unlining